src-electron/generator/helper-zcl.js

```diff
@@ -144,21 +144,36 @@ async function zcl_structs(options) {
     )
   }
   structs = await zclUtil.sortStructsByDependency(structs)
-  structs.forEach((st) => {
+  for (const st of structs) {
     st.struct_contains_array = false
     st.struct_has_fabric_sensitive_fields = false
     st.has_no_clusters = st.struct_cluster_count < 1
     st.has_one_cluster = st.struct_cluster_count == 1
     st.has_more_than_one_cluster = st.struct_cluster_count > 1
-    st.items.forEach((i) => {
-      if (i.isArray) {
-        st.struct_contains_array = true
-      }
-      if (i.isFabricSensitive) {
-        st.struct_has_fabric_sensitive_fields = true
+    const checkTransitive = async (items) => {
+      for (const i of items) {
+        if (i.isArray) {
+          st.struct_contains_array = true
+        }
+        if (i.isFabricSensitive) {
+          st.struct_has_fabric_sensitive_fields = true
```

Comment:
This is wrong, given how this boolean actually gets used.  It really is about the specific fields of this struct, not other things inside it.

In general, having non-toplevel structs with fabric-sensitive fields is not really defined in the spec, and not supported in implementation; if we want to support it we need to figure out how it should work.

Reply:
>  having non-toplevel structs with fabric-sensitive fields is not really defined in the spec, and not supported in implementation

If this is the case, then this is not wrong? i.e. the behavior is undefined, with or without this change?

Reply:
Looking for fabric-sensitive bits in non-toplevel structs and assigning semantics to it means that if someone does add something like this, no red flags will go up to actually define it.

We should not change the fabric-sensitive behavior here.

Reply:
OK cool, fair enough, easy enough to exclude fabric-sensitive (though it does seem like either way it'll be broken)